kcbq-connector/src/main/java/com/wepay/kafka/connect/bigquery/write/row/BigQueryWriter.java

```diff
@@ -102,15 +106,25 @@ public void writeRows(PartitionedTableId table,
     logger.debug("writing {} row{} to table {}", rows.size(), rows.size() != 1 ? "s" : "", table);
 
     Exception mostRecentException = null;
+    Map<Long, List<BigQueryError>> partialFailRowsMap = null;
 
     int retryCount = 0;
     do {
       if (retryCount > 0) {
         waitRandomTime();
       }
       try {
-        performWriteRequest(table, rows, topic);
-        return;
+        partialFailRowsMap = performWriteRequest(table, rows, topic);
+        if (partialFailRowsMap == null || partialFailRowsMap.isEmpty()) {
+          // table insertion completed with no reported errors
+          return;
+        } else {
```

Comment:
We may also want to go through the failed rows and make sure that the error is actually a partial failure error. We can just straight up throw an exception if it's something else, rather than retrying probably futilely.

Reply:
It's not easy to tell a "partial failure" based on `partialFailRowsMap, because there isn't an "error code" in `BigQueryError` object. 
Actually, I think if `bigQuery.insertAll()` does not throw a BigQueryException, meanwhile `writeResponse.hasErrors()`, it is a "partial fail" (consistent with previous stack trace).
What you think?
https://github.com/wepay/kafka-connect-bigquery/pull/108/files#diff-5e819bb0e646d1ad0aa8d0281013ef43R67
https://github.com/wepay/kafka-connect-bigquery/issues/53

Reply:
I think my comment here: https://github.com/wepay/kafka-connect-bigquery/pull/108#discussion_r178650979 addresses this.